lib/tasks/transcribe.js

```diff
@@ -67,10 +75,61 @@ class TaskTranscribe extends Task {
 
   get name() { return TaskName.Transcribe; }
 
+  async _initSpeechCredentials(cs, vendor, label) {
+    const {getNuanceAccessToken, getIbmAccessToken} = this.cs.srf.locals.dbHelpers;
+    let credentials = cs.getSpeechCredentials(vendor, 'stt', label);
+
+    if (!credentials) {
+      const {writeAlerts, AlertType} = cs.srf.locals;
+      this.logger
+        .info(`TaskTranscribe:_initSpeechCredentials - ERROR stt using ${vendor} requested but creds not supplied`);
+      writeAlerts({
+        account_sid: cs.accountSid,
+        alert_type: AlertType.STT_NOT_PROVISIONED,
+        vendor
+      }).catch((err) => this.logger.info({err}, 'Error generating alert for no stt'));
+      // Notify application that STT vender is wrong.
+      this.notifyError({
+        msg: 'ASR error',
+        details: `No speech-to-text service credentials for ${vendor} have been configured`
+      });
+      this.notifyTaskDone();
+      throw new Error(`No speech-to-text service credentials for ${vendor} have been configured`);
+    }
+
+    if (vendor === 'nuance' && credentials.client_id) {
+      /* get nuance access token */
+      const {client_id, secret} = credentials;
+      const {access_token, servedFromCache} = await getNuanceAccessToken(client_id, secret, 'asr tts');
+      this.logger.debug({client_id}, `Gather:exec - got nuance access token ${servedFromCache ? 'from cache' : ''}`);
+      credentials = {...credentials, access_token};
+    }
+    else if (vendor == 'ibm' && credentials.stt_api_key) {
+      /* get ibm access token */
+      const {stt_api_key, stt_region} = credentials;
+      const {access_token, servedFromCache} = await getIbmAccessToken(stt_api_key);
+      this.logger.debug({stt_api_key}, `Gather:exec - got ibm access token ${servedFromCache ? 'from cache' : ''}`);
+      credentials = {...credentials, access_token, stt_region};
+    }
+    return credentials;
+  }
+
+  async _fallback() {
```

Comment:
I see a lot of duplicated code in gather and transcribe.  Of course, that has always been the case.  But I am wondering -- should we create a subclass called TaskStt that inherits from Task, factor all of these functions into that subclass, and then inherit both TaskGather and TaskTranscribe from TaskStt instead of from Task?